File: Python/206. Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        p, ans = head, None
        while p:
            tmp = p.next
            p.next = ans
            ans = p
            p = tmp
        return ans

File: C++/206. Reverse Linked List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* ans = nullptr;
        ListNode* p = head;
        while(p)
        {
            ListNode* tmp = p->next;
            p->next = ans;
            ans = p;
            p = tmp;
        }
        return ans;
    }
};


class Solution2 {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* p = head;
        ListNode* ans = nullptr;
        while(p)
        {
            ListNode* tmp = new ListNode(p->val);
            tmp->next = ans;
            ans = tmp;
            p = p->next;
            if(p == nullptr)
                break;
        }
        return ans;
    }
};